app/decorators/user_decorator.rb

```diff
@@ -18,14 +18,12 @@ class UserDecorator < ApplicationDecorator
     },
   ].freeze
 
-  delegate_all
-
   def cached_followed_tags
     Rails.cache.fetch("user-#{id}-#{updated_at}/followed_tags_11-30", expires_in: 20.hours) do
-      follows_query = Follow.where(follower_id: id, followable_type: "ActsAsTaggableOn::Tag").pluck(:followable_id, :points)
-      tags = Tag.where(id: follows_query.map { |f| f[0] }).order("hotness_score DESC")
+      follows = Follow.where(follower_id: id, followable_type: "ActsAsTaggableOn::Tag").pluck(:followable_id, :points)
+      tags = Tag.where(id: follows.map { |f| f[0] }).order("hotness_score DESC")
       tags.each do |t|
-        follow_query_item = follows_query.detect { |f| f[0] == t.id }
+        follow_query_item = follows.detect { |f| f[0] == t.id }
```

Comment:
If we made follows a hash with pairs like this `{id: points}` then this lookup with be a lot faster bc we wouldn't have to iterate through the list every single time.

Reply:
Done! https://github.com/thepracticaldev/dev.to/pull/6125/commits/790ecd29d0ac4b1ecf45e7a34b04f8bbb9b2df6d